drivers/timer/ti_dmtimer.c

```diff
@@ -0,0 +1,167 @@
+/* Copyright (C) 2023 BeagleBoard.org Foundation
+ * Copyright (C) 2023 S Prashanth
+ * Copyright (c) 2024 Texas Instruments Incorporated
+ *	Andrew Davis <afd@ti.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/timer/system_timer.h>
+#include <zephyr/irq.h>
+#include <zephyr/sys_clock.h>
+#include <zephyr/kernel.h>
+#include <zephyr/spinlock.h>
+
+#include <zephyr/drivers/timer/ti_dmtimer.h>
+
+#define DT_DRV_COMPAT ti_am654_timer
+
+#define TIMER_BASE_ADDR DT_INST_REG_ADDR(0)
+
+#define TIMER_IRQ_NUM   DT_INST_IRQN(0)
+#define TIMER_IRQ_PRIO  DT_INST_IRQ(0, priority)
+#define TIMER_IRQ_FLAGS DT_INST_IRQ(0, flags)
+
+#define CYC_PER_TICK ((uint32_t)(sys_clock_hw_cycles_per_sec() \
+				/ CONFIG_SYS_CLOCK_TICKS_PER_SEC))
+
+#define MAX_TICKS ((k_ticks_t)(UINT32_MAX / CYC_PER_TICK) - 1)
+
+static struct k_spinlock lock;
+
+static uint32_t last_cycle;
+
+#define TI_DM_TIMER_READ(reg) sys_read32(TIMER_BASE_ADDR + TI_DM_TIMER_ ## reg)
+
+#define TI_DM_TIMER_MASK(reg) TI_DM_TIMER_ ## reg ## _MASK
+#define TI_DM_TIMER_SHIFT(reg) TI_DM_TIMER_ ## reg ## _SHIFT
+#define TI_DM_TIMER_WRITE(data, reg, bits) \
+	ti_dm_timer_write_masks(data, \
+		TIMER_BASE_ADDR + TI_DM_TIMER_ ## reg, \
+		TI_DM_TIMER_MASK(reg ## _ ## bits), \
+		TI_DM_TIMER_SHIFT(reg ## _ ## bits))
+
+static void ti_dm_timer_write_masks(uint32_t data, uint32_t reg, uint32_t mask, uint32_t shift)
+{
+	uint32_t reg_val;
+
+	reg_val = sys_read32(reg);
+	reg_val = (reg_val & ~(mask)) | (data << shift);
+	sys_write32(reg_val, reg);
+}
+
+void ti_dmtimer_isr(void *data)
```

Comment:
`static`

Reply:
ACK, will fix that in the DM timer series.